lib/interactive_viewer.ml

```diff
@@ -69,11 +69,29 @@ let view (patches : Patch.t list) =
     | Some z -> Lwd.var z
     | None -> failwith "zipper_of_list: empty list"
   in
+  let curr_scroll_state = Lwd.var W.default_scroll_state in
+  let change_scroll_state action state =
+    let off_screen = action = `Action && state.W.position > state.W.bound in
```

Comment:
Your function do not handle the `` `Content `` action, which is called when the size of the content of the area changes. You need to handle that the `bound` can change in this case.

This creates a bug:
- Go to a long diff hunk (for example, the fourth file in `example.diff`)
- Scroll to the bottom
- Go to a shorter diff hunk (for example, the next one)
- The screen is empty because the scroll is now too far off.